src/EditorFeatures/Core/InheritanceMargin/InheritanceMarginServiceHelpers.cs

```diff
@@ -256,23 +297,40 @@ internal static class InheritanceMarginServiceHelper
         /// Overridden: O↓
         /// Overriding: O↑
         /// </summary>
-        private static async Task<ImmutableArray<ISymbol>> GetImplementedSymbolsAsync(
+        private static async Task<ImmutableArray<ISymbol>> GetImplementedSymbolsForTypeMemberAsync(
             Solution solution,
             ISymbol memberSymbol,
             CancellationToken cancellationToken)
         {
-            if (memberSymbol is INamedTypeSymbol { IsSealed: false } namedTypeSymbol)
-            {
-                var derivedTypes = await GetDerivedTypesAndImplementationsAsync(solution, namedTypeSymbol, cancellationToken).ConfigureAwait(false);
-                return derivedTypes.CastArray<ISymbol>();
-            }
-            else if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol
+            if (memberSymbol is IMethodSymbol or IEventSymbol or IPropertySymbol
                  && memberSymbol.ContainingSymbol.IsInterfaceType())
             {
-                return await SymbolFinder.FindMemberImplementationsArrayAsync(
+                // Use a hashset builder because there could be duplicates.
+                // For example:
+                // interface IBar { void Foo(); }
+                // class Bar : IBar { public virtual void Foo() { } }
+                // class Bar2 : Bar, IBar { public override void Foo() { } }
+                // Here Bar2.Foo() would be added twice as an implemented members.
+                // The order is not important, because UI would sort it based on display name.
+                using var _ = PooledHashSet<ISymbol>.GetInstance(out var builder);
+                // 1. Find all direct implementations for this member
+                var implementationSymbols = await SymbolFinder.FindMemberImplementationsArrayAsync(
                     memberSymbol,
                     solution,
                     cancellationToken: cancellationToken).ConfigureAwait(false);
+                builder.AddRange(implementationSymbols);
+
+                // 2. Continue searching the overriden symbols. For example:
+                // interface IBar { void Foo(); }
+                // class Bar : IBar { public virtual void Foo() { } }
+                // class Bar2 : IBar { public override void Foo() { } }
+                // For 'IBar.Foo()', we need to find 'Bar2.Foo()'
+                foreach (var implementationSymbol in implementationSymbols)
+                {
+                    builder.AddRange(await SymbolFinder.FindOverridesArrayAsync(implementationSymbol, solution, cancellationToken: cancellationToken).ConfigureAwait(false));
+                }
+
+                return builder.ToImmutableArray();
```

Comment:
so isntead of doing all the hasing/distincting here.  why not just have hte logic here not care, and have the final collapsing done at the highest layer

Reply:
Done